lib/_http_client.js

```diff
@@ -801,7 +801,7 @@ function onSocketNT(req, socket, err) {
       req.emit('close');
     }
 
-    if (!err && req.agent) {
+    if (!err && req.agent && !socket.destroyed) {
```

Comment:
@lpinca: This is the primary fix which also ensure the error is forwarded and not swallowed.